src/WinRT.Runtime/Projections/IPropertyValue.net5.cs

```diff
@@ -325,12 +326,46 @@ private static T CoerceValue<T>(object value)
                 {
                     return (T)(object)guid.ToString("D", global::System.Globalization.CultureInfo.InvariantCulture);
                 }
+                else if (typeof(T) == typeof(byte))
+                {
+                    return (T)(object)Convert.ToByte(value, global::System.Globalization.CultureInfo.InvariantCulture);
+                }
+                else if (typeof(T) == typeof(short))
+                {
+                    return (T)(object)Convert.ToInt16(value, global::System.Globalization.CultureInfo.InvariantCulture);
+                }
+                else if (typeof(T) == typeof(ushort))
+                {
+                    return (T)(object)Convert.ToUInt16(value, global::System.Globalization.CultureInfo.InvariantCulture);
+                }
+                else if (typeof(T) == typeof(int))
+                {
+                    return (T)(object)Convert.ToInt32(value, global::System.Globalization.CultureInfo.InvariantCulture);
+                }
+                else if (typeof(T) == typeof(uint))
+                {
+                    return (T)(object)Convert.ToUInt32(value, global::System.Globalization.CultureInfo.InvariantCulture);
+                }
+                else if (typeof(T) == typeof(long))
+                {
+                    return (T)(object)Convert.ToInt64(value, global::System.Globalization.CultureInfo.InvariantCulture);
+                }
+                else if (typeof(T) == typeof(ulong))
+                {
+                    return (T)(object)Convert.ToUInt64(value, global::System.Globalization.CultureInfo.InvariantCulture);
+                }
+                else if (typeof(T) == typeof(float))
+                {
+                    return (T)(object)Convert.ToSingle(value, global::System.Globalization.CultureInfo.InvariantCulture);
+                }
+                else if (typeof(T) == typeof(double))
+                {
+                    return (T)(object)Convert.ToDouble(value, global::System.Globalization.CultureInfo.InvariantCulture);
+                }
                 else
```

Comment:
We do also call this with `DateTimeOffset` and `TimeSpan`.  Should those be handled here too?

https://github.com/microsoft/CsWinRT/blob/7bdf00b08d6a4d5f4554148d904e9c5d2f0aa00c/src/WinRT.Runtime/Projections/IPropertyValue.net5.cs#L614C1-L645

Reply:
DateTimeOffset and TimeSpan have never been supported in this scenario (we never coerce to them, we only support when they're already correctly-typed). I'd rather not add new functionality here unless we get an ask to do so.

Reply:
So in those 2 cases, it will just be handled by the initial if that does `value is T u` right?

Reply:
Actually I noticed that the block here was limited to NumericScalarTypes so this is equivalent to before.